eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/BlacklistedWordsCheck.java

```diff
@@ -60,12 +61,22 @@ public void visitToken(DetailAST token) {
             case TokenTypes.CLASS_DEF:
             case TokenTypes.METHOD_DEF:
             case TokenTypes.VARIABLE_DEF:
-                if (isPublicApi(token)) {
-                    String tokenName = token.findFirstToken(TokenTypes.IDENT).getText();
-                    if (hasBlacklistedWords(tokenName)) {
-                        log(token, String.format(ERROR_MESSAGE, tokenName, this.blacklistedWords.stream().collect(Collectors.joining(", ", "", ""))));
-                    }
+                if (!isPublicApi(token)) {
+                    break;
                 }
+
+                final String tokenName = token.findFirstToken(TokenTypes.IDENT).getText();
+                if (!hasBlacklistedWords(tokenName)) {
+                    break;
+                }
+
+                // In an interface all the fields (variables) are by default public, static and final.
+                if (token.getType() == TokenTypes.VARIABLE_DEF && ScopeUtil.isInInterfaceBlock(token)) {
+                    break;
+                }
+
+                log(token, String.format(ERROR_MESSAGE, tokenName, String.join(", ", this.blacklistedWords)));
```

Comment:
nit: Should we make the error message a variable in the Checkstyle instead of performing a format on every error.